### ISSUE TITLE
Improve how we track platform events

### ISSUE THREAD

**Issue by gosusnp** [airbytehq/airbyte#19522]
## Tell us about the problem you're trying to solve
Currently, the platform is pre-aggregating some events before emitting them for analytics/reporting.
This is leading to some issues:
* We can be losing events in some cases, for example in the case of an OOM during a sync, our logs doesn't show the sync because we report at the end of the sync
* Some granularity is lost which means that refining analysis may involve changing how the platform is tracking/emitting the data

For example, trying to dive into failures of our syncs, we are missing some data that syncs started.

## Describe the solution you’d like
The platform should strive to emit simple events as soon as possible. If we have some rate limit concerns, batching event is fine, ideally, batching should be done at most at a minute.
* simple events each operation, for example: replication_start, replication_stop, normalization_start, normalization_stop, etc, reset_start, reset_stop, *_failures
* events should have the right keys to be able to join them
  * workspace_id, connection_id, attempt_id
  * we need to be sure to be able to tie any events to a workspace/connection, as well as have enough information to be able to build a timeline for a sync: for one attempt, [check_connection_start, check_connection_success, replication_start, ...]

## Describe the alternative you’ve considered or used
Open to suggestions

**Comment by alex-gron** [comment]
@gosusnp Thank you! This is a great start! 

A few comments/questions: 

- We will also need `job_id` (In addition to workspace_id, connection_id, and attempt_id)
- Can we send state messages as part of the event stream?
-  Is this for both OSS and Cloud? 

@ChristopheDuong anything else you want to add here?

**Comment by ChristopheDuong** [comment]
> @ChristopheDuong anything else you want to add here?

Your comment just makes me think that I wished we could have proper context management on events being tracked:
https://www.iglooanalytics.com/blog/understanding-snowplow-analytics-custom-contexts.html 

anyway, do think adding the catalog used at the time of that sync would be possible too? so we know how many streams in the contest of that particular job and what were their settings

**Comment by gosusnp** [comment]
@alex-gron, yes this could be for both, however, I am not sure how the OSS part is leveraged today.
For the state messages, I think we should verify the volume due to the size of some state messages, I am thinking about CDC for example. If the main goal to track checkpointing, I believe we could do something by sending hashes instead of the actual state.

@ChristopheDuong, similar concern about the size of the catalogs. I imagine we could try to handle sync catalog the same way we handle source catalogs today which would solve the size issue.
